### PR TITLE
Implement file compression

### LINKED ISSUE #59: Add support for file compression
Compressing files before uploading them would be a good idea to reduce storage cost.

### PR DESCRIPTION
**Pull request by juslar**
Fix #59

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -59,6 +60,12 @@ export class Datastore {
     async addFile(name: string, publicStatus: boolean, file: ArrayBuffer) {
         await this._initialize()
 
+        if (JSZip.support.arraybuffer) {
```
Comment (macor161) [inline src/index.ts]: I see a potential problem here. What if a user uploads a file with a browser that doesn't support arraybuffer compression? It'll throw an error every time other users try to get the file if they have a browser that supports arraybuffer compression. Either we specify somewhere that the file is compressed or not, or we simply don't allow storing files if the browser doesn't support arraybuffer compression.

File: src/index.ts [hunk 2]
```diff
@@ -59,6 +60,12 @@ export class Datastore {
...
+            let zip = JSZip()
+            await zip.file(name, file)
+            file = await zip.generateAsync({type : "arraybuffer"})
+        }
+
         let encryptionKey = ""
         let storageId
         if (!publicStatus) {
```
Comment (macor161) [inline src/index.ts:75]: I can't comment on lines 75 and 78. What do you think if we use the `byteLength` of the uncompressed file instead?

File: src/index.ts [hunk 3]
```diff
@@ -270,6 +289,11 @@ export class Datastore {
 
         let storageId = ""
         let file = await this.getFile(fileId)
+        if (JSZip.support.arraybuffer) {
```
Comment (macor161) [inline src/index.ts]: Not exactly sure why we're doing this?

**Review by macor161 — CHANGES_REQUESTED** [review]
Nice work!! This will improve our storage efficiency!

File: src/index.ts (hunk 3 above)
Comment (juslar) [reply src/index.ts]: `getFile()` returns the file uncompressed. We need to re-compress it.

File: src/index.ts (hunk 2 above)
Comment (juslar) [reply src/index.ts:75]: Yup! Will do

File: src/index.ts (hunk 1 above)
Comment (juslar) [reply src/index.ts]: Good catch. I now throw an exception if the browser doesn't support JSZip.

File: src/index.ts (hunk 3 above)
Comment (macor161) [reply src/index.ts]: You are absolutely right! My bad! :)

**Review by macor161 — APPROVED** [review]

**Comment by macor161** [comment]
Great job!

**Pull request merged**